src/init.cpp

```diff
@@ -911,9 +911,9 @@ bool AppInitParameterInteraction()
         InitWarning(strprintf(_("Reducing -maxconnections from %d to %d, because of system limitations."), nUserMaxConnections, nMaxConnections));
 
     // ********************************************************* Step 3: parameter-to-internal-flags
-    if (gArgs.IsArgSet("-debug") > 0) {
+    if (gArgs.IsArgSet("-debug")) {
         // Special-case: if -debug=0/-nodebug is set, turn off debugging messages
-        const std::vector<std::string>& categories = gArgs.GetArgs("-debug");
+        const std::vector<std::string> categories = gArgs.GetArgs("-debug");
```

Comment:
Because https://github.com/bitcoin/bitcoin/pull/9494#discussion_r114871456 and I don't care either way for the reasons @sipa explained.